parlai/agents/transformer/modules.py

```diff
@@ -42,13 +46,16 @@ def _build_encoder(opt, dictionary, embedding=None, padding_idx=None, reduction=
         padding_idx=padding_idx,
         learn_positional_embeddings=opt.get('learn_positional_embeddings', False),
         embeddings_scale=opt['embeddings_scale'],
-        reduction=reduction,
+        reduction_type=reduction_type,
         n_positions=n_positions,
+        n_segments=n_segments,
+        activation=opt['activation'],
```

Comment:
related question -- is this going to break backwards compatiibilty? i wonder if we'll see opt loading issues here ...

Reply:
so new runs should have the default arg which is `'relu'`.

Reply:
It's worth noting we have a unit test for transformer backwards compatibility.

Reply:
cool